src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala

```diff
@@ -72,8 +74,47 @@ abstract class SpecializeTypes extends InfoTransform with TypingTransformers {
   override def changesBaseClasses = true
   override def keepsTypeParams = true
 
-  type TypeEnv = immutable.Map[Symbol, Type]
-  def emptyEnv: TypeEnv = Map[Symbol, Type]()
+  final case class TypeEnvWrapper(value: immutable.Map[Symbol, Type]) {
```

Comment:
Why is `TypeEnvWrapper` a case class, instead of a class that extends from `AnyVal`?